projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget.model.ts

```diff
@@ -112,6 +113,14 @@ export class CartesianWidgetModel<TData> {
   })
   public legendPosition: LegendPosition = LegendPosition.TopRight;
 
+  @ModelProperty({
+    key: 'max-data-points',
+    required: false,
+    displayName: 'Maximum Data Points',
+    type: NUMBER_PROPERTY.type
+  })
+  public maxDataPoints?: number;
```

Comment:
We are limiting at widget level. Does it mean 150 points per series? or overall?

Reply:
I'd expect it to be per series, maybe a new name like maxSeriesDataPoints

Reply:
Yes.